src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNetWebApi2Integration.cs

```diff
@@ -11,30 +16,47 @@ public static class AspNetWebApi2Integration
     {
         internal const string OperationName = "aspnet-web-api.request";
 
+        private static readonly Type HttpControllerContextType = Type.GetType("System.Web.Http.Controllers.HttpControllerContext, System.Web.Http", throwOnError: false);
+
         /// <summary>
         /// ExecuteAsync calls the underlying ExecuteAsync and traces the request.
         /// </summary>
         /// <param name="this">The Api Controller</param>
         /// <param name="controllerContext">The controller context for the call</param>
         /// <param name="cancellationTokenSource">The cancellation token source</param>
         /// <returns>A task with the result</returns>
-        public static dynamic ExecuteAsync(dynamic @this, dynamic controllerContext, dynamic cancellationTokenSource)
+        public static object ExecuteAsync(object @this, object controllerContext, object cancellationTokenSource)
         {
             using (Scope scope = CreateScope(controllerContext))
             {
                 return scope.Span.Trace(
-                    () => @this.ExecuteAsync(controllerContext, ((CancellationTokenSource)cancellationTokenSource).Token),
```

Comment:
The method lookup should be done outside of the trace. Can you move that to before the `using` block and just invoke `executeAsyncFunc` here?